gstreamer/src/date_time_serde.rs

```diff
@@ -7,53 +7,80 @@
 // except according to those terms.
 
 use serde::de::{Deserialize, Deserializer};
-use serde::ser::{Serialize, SerializeStruct, Serializer};
+use serde::ser;
+use serde::ser::{Serialize, Serializer};
 use DateTime;
 
+#[derive(Serialize, Deserialize)]
+enum DateTimeVariants {
+    Y(i32),
+    YM(i32, i32),
+    YMD(i32, i32, i32),
+    YMDhm(i32, i32, i32, i32, i32),
+    YMDhmsTz(i32, i32, i32, i32, i32, f64, f32),
```

Comment:
This seems not correct, from looking at the `GstDateTime` code:

```c
typedef enum
{
  GST_DATE_TIME_FIELDS_INVALID = 0,
  GST_DATE_TIME_FIELDS_Y,       /* have year                */
  GST_DATE_TIME_FIELDS_YM,      /* have year and month      */
  GST_DATE_TIME_FIELDS_YMD,     /* have year, month and day */
  GST_DATE_TIME_FIELDS_YMD_HM,
  GST_DATE_TIME_FIELDS_YMD_HMS
      /* Note: if we ever add more granularity here, e.g. for microsecs,
       * the compare function will need updating */
} GstDateTimeFields;
```

It seems that timezone is allowed regardless of everything else. So maybe we can add that simply as an `Option` into each variant?

Reply:
Ok, this was not clear to me. With the `new_y`, `new_ym` & `new_ymd` there is no argument for the time zone. Should I use the `new` with all arguments version and assign `-1` to the missing parts? What should we do in case the time zone is not indicated? I know I said I would use the local time zone, but the available methods are kind of limited in that regard.

Reply:
It uses `0.0` as timezone elsewhere for that

Reply:
Looking at `gst_date_time_get_time_zone_offset`, it seems that the time zone is only meaningful when `DateTime` holds time:

``` C
gst_date_time_get_time_zone_offset (const GstDateTime * datetime)
{
  g_return_val_if_fail (datetime != NULL, 0.0);
  g_return_val_if_fail (gst_date_time_has_time (datetime), 0.0);
  [...]
}
```

That would explain why the `new_y`, `new_ym` and `new_ymd` don't allow defining the time zone. If you confirm my understanding, I'll just add the `Option<f32>` to the `YMDhm` and `YMDhms` variants.

Reply:
Confirmed and agreed about no option